api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -1141,7 +1141,19 @@ public ConceptSource getConceptSourceByName(String conceptSourceName) throws DAO
 		criteria.add(Restrictions.eq("source.name", conceptSourceName));
 		return (ConceptSource) criteria.uniqueResult();
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.ConceptDAO#getConceptSourceByUniqueId(java.lang.String)
+	 */
+	public ConceptSource getConceptSourceByUniqueId(String uniqueId) {
+		if (StringUtils.isBlank(uniqueId)) {
+			return null;
```

Comment:
My line of thinking was that, why would one ask for a concept source with a null or empty parameter? Wouldn't this be a mistake that they should get an exception about?